spec/models/regular_transaction_spec.rb

```diff
@@ -97,4 +97,28 @@
       expect(records).to contain_exactly(debit_payment)
     end
   end
+
+  describe ".frequencies_for" do
+    context "when transaction type is benefits" do
+      it "does not include monthly in the frequencies" do
+        benefits = create(:transaction_type, :benefits)
+
+        frequencies = described_class.frequencies_for(benefits)
+
+        expect(frequencies.count).to eq(4)
+        expect(frequencies).not_to include("monthly")
+      end
+    end
+
+    context "when transaction type is not benefits" do
+      it "includes monthly in the frequencies" do
+        child_care = create(:transaction_type, :child_care)
+
+        frequencies = described_class.frequencies_for(child_care)
+
+        expect(frequencies.count).to eq(5)
+        expect(frequencies).to include("monthly")
+      end
+    end
+  end
```

Comment:
better to use rspec DSL, and to be explicit about the values rather than just counting IMO
```
describe ".frequencies_for" do
  subject(:frequencies_for) { described_class.frequencies_for(transaction_type) }

  context "when transaction type is benefits" do
    let(:transaction_type) { create(:transaction_type, :benefits) }

    it "includes all but monthly in the frequencies" do
      expect(frequencies_for).to match(%w[weekly two_weekly four_weekly three_monthly])
    end
  end

  context "when transaction type is not benefits" do
    let(:transaction_type) { create(:transaction_type, :child_care) }

    it "includes all frequencies" do
      expect(frequencies_for).to match(%w[weekly two_weekly four_weekly monthly three_monthly])
    end
  end
  ```